Support multiple device type with primary type

aspecter opened the pull request.

aspecter committed:
Support multiple device type with primary input

aspecter committed:
Support multiple device type with primary input

aspecter committed:
Support multiple device type with array input

aspecter committed:
Support multiple device type with array input

brdandu committed:
Enabling multiple zcl device types per endpoint draft commit. Edit this comment later
Github: ZAP#862

brdandu committed:
Enabling multiple zcl device types per endpoint draft commit. Edit this comment later
Github: ZAP#862

paulr34 commented:
@brdandu @tecimovic @aspecter what about Zigbee?

aspecter committed:
Unify array usage for DeviceType and DeviceIndentifier

aspecter committed:
Unify array usage for DeviceType and DeviceIndentifier

paulr34 commented:
Primary device types are defined, right? How would the GUI handle this? How would a user know which device types to include in the subset of the superset? They would look at the spec? 

Is the plan to allow any device type combo and just let the user mess up?

brdandu committed:
Updating updateEndpoint and duplicateEndpoint based on the schema change with the introduction of Endpoint_type_device table
Github: ZAP#682

brdandu committed:
Updating updateEndpoint and duplicateEndpoint based on the schema change with the introduction of Endpoint_type_device table
Github: ZAP#682

brdandu committed:
Updating import/export endpoint type apis to handle multiple zcl device types per endpoint
Github: ZAP#862

brdandu committed:
Updating import/export endpoint type apis to handle multiple zcl device types per endpoint
Github: ZAP#862

brdandu committed:
Updating selectAllEndpointTypes and selectEndpointType such that they update the multiple zcl device types correctly
Github: ZAP#862

brdandu committed:
Updating selectAllEndpointTypes and selectEndpointType such that they update the multiple zcl device types correctly
Github: ZAP#862

brdandu committed:
Minor cleanup
Github: ZAP#862

brdandu committed:
Minor cleanup
Github: ZAP#862

aspecter committed:
Add primary device type selection logic

aspecter committed:
Add primary device type selection logic

aspecter committed:
Add multi device type modification warning dialog

aspecter committed:
Add multi device type modification warning dialog

aspecter committed:
Generalize the warning message

aspecter committed:
Generalize the warning message

brdandu committed:
Maintain specific device type ref name, device type code and device type profile id per endpoint for backwards compatibility
Github: ZAP#862

brdandu committed:
Removing the warning dialog and making sure the the zcl device types per endpoint are updated correctly
Github: ZAP#862

brdandu committed:
Minor cleanup
Github: ZAP#862

brdandu committed:
Removing the UI pieces for this and the tests since this is not really useful in any way
Github: ZAP#862

brdandu committed:
Making sure that the device type clusters are added when an endpoint is edited and a new zcl device type is added to the endpoint
Github: ZAP#862

brdandu committed:
When loading the old zap files and editing the endpoint the selected devices do not show up in the device UI. Updating deviceIdentifier with this.zclDeviceTypes[deviceTypeRefs[i]].code fixed it
Github: ZAP#862

aspecter committed:
Merge remote-tracking branch 'brdandu/feature/addMultipleDeviceTypesPerEndpoint/ZAP#862' into multiple-device-type

aspecter committed:
Text adjustment to the new design

aspecter committed:
Remove warning dialog code chunks

brdandu committed:
Add device type identifier and device type revision to endpoint_type_device junction table instead of keeping it in endpoint table such that multiple zcl device types can be added per endpoint. Schema change lead to the rest of the changes in queries for endpoint_type, endpoint_type_device and endpoint tables along with UI changes to vue files, actions and mutations.
Github: ZAP#862

aspecter committed:
Merge remote-tracking branch 'brdandu/feature/addMultipleDeviceTypesPerEndpoint/ZAP#862' into multiple-device-type

aspecter committed:
Add multi version option to endpoint card

aspecter committed:
Add multi version option to endpoint card

brdandu committed:
Updating device version and device identifier to be in endpoint type instead of endpoint just like device type ref

- Updating db-mapping to have device version and identifier in endpoint_type
- Fixing query-config#updateEndpointType to add the entries properly to endpoint_type and endpoint_type_device tables
- Use endpoint_type_device table in a bunch of queries instead of endpoint_type to get references to device version and identifier
- Fix import isc with the new schema
- Fix edit and update endpoint by fixing httpPatchEndpointType with respect to device version and identifier
- Fix Create endpoint UI and endpoint card such that device identifier and device version can be updated properly through the UI
- Fixing actions and mutations for device version and identifier for an endpoint type such that UI and backend are in sync
- Updating state.js such that device version and identifier are in endpoint type instead of endpoint now
- Minor cleanup
- Github: ZAP#862

brdandu committed:
Fix import issues with the new schema for multiple zcl device types and device version/identifier
Github: ZAP#862

brdandu committed:
Fixing import and export of .zap files such that saving and re-opening of zap files continues to give the right results when multiple zcl device types are chosen
Github: ZAP#862

aspecter committed:
Merge branch 'feature/addMultipleDeviceTypesPerEndpoint/ZAP#862' into multiple-device-type

aspecter committed:
Update Cypress tests to support multi device types

aspecter committed:
Update Cypress tests to support multi device types

brdandu committed:
Adding tests for adding multiple zcl device types per endpoint. The tests include adding a new endpoint and editing an existing endpoint
Github: ZAP#862

aspecter committed:
Merge commit 'e360bc1ae518c3ce90d66037f589550a96f66a7c' into multiple-device-type

aspecter committed:
Make device select show selected item at edit mode
